packages/scene-composer/src/components/panels/scene-rule-components/SceneRuleTargetEditor.tsx

```diff
@@ -92,23 +102,47 @@ export const SceneRuleTargetEditor: React.FC<ISceneRuleTargetEditorProps> = ({
           <SceneRuleTargetIconEditor
             targetValue={targetInfo.value}
             onChange={(targetValue) => {
-              const colorWithIcon = targetValue === 'Custom' ? `${targetValue}-${chosenColor}` : targetValue;
-              onChange(convertToIotTwinMakerNamespace(targetInfo.type, colorWithIcon));
+              onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetValue));
             }}
             chosenColor={chosenColor}
+            customIcon={icon as IconLookup}
           />
           {isCustomStyle && (
-            <ColorPicker
-              color={chosenColor}
-              onSelectColor={(newColor) => {
-                const colorWithIcon =
-                  targetInfo.value === 'Custom' ? `${targetInfo.value}-${newColor}` : targetInfo.value;
-                onChange(convertToIotTwinMakerNamespace(targetInfo.type, colorWithIcon));
-                setChosenColor(newColor);
-              }}
-              colorPickerLabel={formatMessage({ defaultMessage: 'Colors', description: 'Colors' })}
-              customColorLabel={formatMessage({ defaultMessage: 'Custom colors', description: 'Custom colors' })}
-            />
+            <>
+              <ColorPicker
+                color={chosenColor}
+                onSelectColor={(newColor) => {
+                  onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetInfo.value), {
+                    ...targetMetadata,
+                    color: newColor,
+                  } as TargetMetadata<string>);
+                  setChosenColor(newColor);
+                }}
+                customColors={tagStyleColors}
+                onUpdateCustomColors={(customColors) =>
+                  setSceneProperty(KnownSceneProperty.TagCustomColors, customColors)
+                }
+                colorPickerLabel={formatMessage({ defaultMessage: 'Colors', description: 'Colors' })}
+                customColorLabel={formatMessage({ defaultMessage: 'Custom colors', description: 'Custom colors' })}
+              />
+              <IconPicker
+                onSelectIconChange={(pickedIcon) => {
+                  onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetInfo.value), {
+                    ...targetMetadata,
+                    iconPrefix: pickedIcon.prefix,
+                    iconName: pickedIcon.iconName,
+                  } as TargetMetadata<string>);
+                  setIcon(pickedIcon as IconLookup);
+                }}
+                selectedIcon={icon as IconLookup}
+                iconPickerLabel={formatMessage({ defaultMessage: 'Icon', description: 'Icon' })}
+                iconFilterText={formatMessage({ defaultMessage: 'Find icons', description: 'Find icons' })}
+                iconFilterTextAriaLabel={formatMessage({
+                  defaultMessage: 'Filter icons',
+                  description: 'Filter icons',
+                })}
+              />
+            </>
```

Comment:
all of those are specific to icon target, so they should be in `SceneRuleTargetIconEditor` component instead of here. and all states related to chosenColor and customIcon should all be move into `SceneRuleTargetIconEditor`

Reply:
They are specific to Icon and logic is within it. There are two reasons why we need to add them here. 
1. Because of the current grid pattern. There is an outer most grid for targets.
2. There is grid within target components itself
If we have move them the inside icon component then we will have change the UX design as there is a outer grid structure which set the icon component to render from colSpan 4 If you add them part of the icon component. 
I would like to keep as it is for this release unless there is an urgent change in UI requirement.